File: neetcode_roadmap/solutions/Arrays & Hashing/242.py
# https://leetcode.com/problems/valid-anagram/description/

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        nums = [0] * 26
        for i in s:
            nums[ord(i) - ord('a')] += 1
        for i in t:
            nums[ord(i) - ord('a')] -= 1

        for i in range(len(nums)):
            if nums[i] != 0:
                return False
        return True

File: neetcode_roadmap/solutions/Arrays & Hashing/242.cpp
//https://leetcode.com/problems/valid-anagram/

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        vector<int> vec(26);
        for(int i = 0; i < s.size(); i++) {
            vec[s[i] - 'a']++;
            vec[t[i] - 'a']--;
        }
        for (int n : vec) {
            if (n != 0) {
                return false;
            }
        }
        return true;
    }
};